client/data/emails/use-remove-titan-mailbox-mutation.ts

```diff
@@ -73,15 +73,15 @@ export function useRemoveTitanMailboxMutation(
 		} );
 	};
 
-	return useMutation(
-		() =>
+	return useMutation( {
+		mutationFn: () =>
 			wp.req.get( {
 				path: `/emails/titan/${ encodeURIComponent( domainName ) }/mailbox/${ encodeURIComponent(
 					mailboxName
 				) }`,
 				method: 'DELETE',
 				apiNamespace: 'wpcom/v2',
 			} ),
-		mutationOptions
-	);
+		...mutationOptions,
```

Comment:
Theoretically, someone could pass a `mutationFn` to `mutationOptions ` and inadvertently overwrite the one we specify here. Should we move `...mutationOptions` above `mutationFn`?

Alternatively, we could alter the type to use `Omit` and omit the `mutationFn` property like in the other areas in this PR.

Reply:
Added https://github.com/Automattic/wp-calypso/pull/77602/commits/0bc21fb6042781cec3db6f403722ce4b7903bfb9 which takes care of preventing accidental `mutationFn` overrides.